### PR TITLE
feat(snapshots): include chunkids in snapshot checksum

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* Include chunkid when calculating checksum of a snapshot and snapshot chunks
* Update how the checksum is calculated. Instead of adding the checksums together, use the same checksum object and update it with new contents.

## Related issues

Related #6666

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@npepinpe The PR to read the file in chunks will follow.

File: snapshot/src/main/java/io/zeebe/snapshots/impl/SnapshotChunkUtil.java [hunk 1]
```diff
@@ -9,18 +9,34 @@
...
 
 final class SnapshotChunkUtil {
 
   private SnapshotChunkUtil() {}
 
-  static long createChecksum(final byte[] content) {
-    final CRC32C crc32 = new CRC32C();
-    crc32.update(content);
-    return crc32.getValue();
+  static long createChecksum(final String chunkName, final byte[] content) {
```
Comment (npepinpe) [inline snapshot/src/main/java/io/zeebe/snapshots/impl/SnapshotChunkUtil.java:22]: Does this mean the individual chunk checksum now also include the filename? I think it's only necessary when computing the folder checksum - if we just want the file checksum the filename isn't necessary to include, imo.

Comment (deepthidevaki) [reply snapshot/src/main/java/io/zeebe/snapshots/impl/SnapshotChunkUtil.java:22]: I was not sure if we wanted it. I thought we discussed it to unify the checksum calculations. I will revert these.

File: snapshot/src/test/java/io/zeebe/snapshots/impl/SnapshotChecksumTest.java [hunk 2]
```diff
@@ -53,6 +53,22 @@ public void shouldGenerateTheSameChecksumForOneFile() throws Exception {
...
+  @Test
+  public void shouldGenerateDifferentChecksumWhenFileNameIsDifferent() throws Exception {
+    // given
+    final var expectedChecksum = SnapshotChecksum.calculate(singleFileSnapshot);
+
+    // when
+    singleFileSnapshot
+        .resolve("file1.txt")
+        .toFile()
+        .renameTo(singleFileSnapshot.resolve("renamed").toFile());
```
Comment (npepinpe) [inline snapshot/src/test/java/io/zeebe/snapshots/impl/SnapshotChecksumTest.java:65]:
```suggestion
    Files.move(singleFileSnapshot.resolve("file1.txt"), singleFileSnapshot.resolve("renamed"));
```

**Review by npepinpe — APPROVED** [review]
:rocket: 

Only one small suggestion but not a blocker

**Pull request merged**